src/logger/sampleRecord.c

```diff
@@ -9,15 +9,65 @@
 #include "mem_mang.h"
 #include "FreeRTOS.h"
 #include "taskUtil.h"
+#include "loggerSampleData.h"
 
-ChannelSample * create_channel_sample_buffer(LoggerConfig *loggerConfig, size_t channelCount)
+#include <stdbool.h>
+
+size_t init_sample_buffer(struct sample *s, const size_t count)
+{
+        if (s->channel_samples)
+                free_sample_buffer(s);
+
+        const size_t size = sizeof(ChannelSample[count]);
+        s->ticks = 0;
+        s->channel_count = count;
+        s->channel_samples = (ChannelSample *) portMalloc(size);
+
```

Comment:
It would seem that we don't.  I'll add a check so that if the malloc fails we will return size 0